rust/src/transformations/make_stable_expr/namespace_dt/expr_datetime_component/test.rs

```diff
@@ -0,0 +1,138 @@
+use std::str::FromStr;
+
+use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+
+use crate::domains::{AtomDomain, DatetimeDomain, LazyFrameDomain, SeriesDomain};
+use crate::metrics::SymmetricDistance;
+use crate::transformations::make_stable_lazyframe;
+
+use super::*;
+use TemporalFunction::*;
+
+#[test]
+fn test_make_expr_components() -> Fallible<()> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("time", AtomDomain::<NaiveTime>::default()),
+        SeriesDomain::new(
+            "datetime",
+            DatetimeDomain {
+                time_unit: TimeUnit::Milliseconds,
+                time_zone: None,
+            },
+        ),
+        SeriesDomain::new("date", AtomDomain::<NaiveDate>::default()),
+    ])?;
+
+    let data = df!(
+        "time" => &[NaiveTime::from_str("00:01:02").unwrap()],
+        "datetime" => &[NaiveDateTime::from_str("2000-01-02T03:04:05").unwrap()],
+        "date" => &[NaiveDate::from_str("2000-01-02").unwrap()]
+    )?
+    .lazy();
+
+    let plan = data.clone().with_columns([
+        col("time").dt().hour().alias("time-hour"),
+        col("datetime").dt().ordinal_day().alias("datetime-day"),
+        col("date").dt().quarter().alias("date-quarter"),
+    ]);
+    let t_components = make_stable_lazyframe(lf_domain, SymmetricDistance, plan)?;
+
+    let actual = t_components.invoke(&data)?.collect()?;
+    let expected = r#"shape: (1, 6)
+┌──────────┬─────────────────────┬────────────┬───────────┬──────────────┬──────────────┐
+│ time     ┆ datetime            ┆ date       ┆ time-hour ┆ datetime-day ┆ date-quarter │
+│ ---      ┆ ---                 ┆ ---        ┆ ---       ┆ ---          ┆ ---          │
+│ time     ┆ datetime[ms]        ┆ date       ┆ i8        ┆ i16          ┆ i8           │
+╞══════════╪═════════════════════╪════════════╪═══════════╪══════════════╪══════════════╡
+│ 00:01:02 ┆ 2000-01-02 03:04:05 ┆ 2000-01-02 ┆ 0         ┆ 2            ┆ 1            │
+└──────────┴─────────────────────┴────────────┴───────────┴──────────────┴──────────────┘"#;
+    assert_eq!(format!("{:?}", actual), expected);
+    Ok(())
+}
+
+fn test_temporal_op_schema<const L: usize>(df: DataFrame, ops: [TemporalFunction; L]) {
```

Comment:
Slight preference to reserve `test_` for tests, and a helper like this might be 
```suggestion
fn assert_temporal_op_schema<const L: usize>(df: DataFrame, ops: [TemporalFunction; L]) {
```

Imagining something looking at the code for a little bit and at first wondering where the `#[test]` is.